pallets/committee/src/lib.rs

```diff
@@ -492,42 +444,27 @@ impl<T: Trait<I>, I: Instance> Module<T, I> {
     }
 
     /// Accepts or rejects the proposal if its threshold is satisfied.
-    fn check_proposal_threshold(proposal: T::Hash) {
-        if let Some(voting) = Self::voting(&proposal) {
+    fn execute_if_passed(proposal: T::Hash) {
+        let voting = match Self::voting(&proposal) {
             // Make sure we don't have an expired proposal at this point.
-            if let Err(_) = Self::ensure_not_expired(
-                &proposal,
-                voting.expiry,
-                system::Module::<T>::block_number(),
-            ) {
-                return;
-            }
-
-            let ayes = voting.ayes.len() as MemberCount;
-            let nays = voting.nays.len() as MemberCount;
-            let did = Context::current_identity::<Identity<T>>().unwrap_or_default();
-            Self::check_threshold_finalize(proposal, did, voting, ayes, nays, Self::seats());
-        }
-    }
+            Some(v) if Self::ensure_not_expired(&proposal, v.expiry).is_ok() => v,
+            _ => return,
+        };
+        let ayes = voting.ayes.len() as MemberCount;
```

Comment:
Maybe it should not be fixed in this ticket:
It is loading unbounded vector just for counting. We could split this into 2 structures:
- PolymesVotes would only stores the count of `ayes` and `nayes` as `u32`.
- A new double map ( (ProposalIndex, Identity) -> Yes/No) to manage the vote.

Reply:
We decided on a meeting to consider this for later.